server/src/grpc/meta_event_service/mod.rs

```diff
@@ -249,21 +253,42 @@ async fn handle_open_shard(ctx: HandlerContext, request: OpenShardRequest) -> Re
             shard_id: shard_info.id,
             cluster_version: topology.cluster_topology_version,
         };
-        schema
-            .open_table(open_request.clone(), opts.clone())
-            .await
-            .box_err()
-            .with_context(|| ErrWithCause {
-                code: StatusCode::Internal,
-                msg: format!("fail to open table, open_request:{open_request:?}"),
-            })?
-            .with_context(|| ErrNoCause {
-                code: StatusCode::Internal,
-                msg: format!("no table is opened, open_request:{open_request:?}"),
-            })?;
+        let result = schema.open_table(open_request.clone(), opts.clone()).await;
+
+        match result {
+            Ok(Some(_)) => {
+                success += 1;
+            }
+            Ok(None) => {
+                error!("no table is opened, open_request:{open_request:?}");
+                err_map.insert(table.name, "no table is opened");
+            }
+            Err(e) => {
+                error!("fail to open table, open_request:{open_request:?}, err:{e}");
+                err_map.insert(table.name, "fail to open table");
+            }
+        };
     }
 
-    Ok(())
+    info!(
+        "Open shard finish, shard id:{}, cost:{}ms, successful tables:{}, failed tables:{}",
+        shard_info.id,
+        instant.saturating_elapsed().as_millis(),
+        success,
+        err_map.len(),
+    );
+
+    if err_map.is_empty() {
+        Ok(())
+    } else {
+        Err(Error::ErrNoCause {
+            code: StatusCode::Internal,
+            msg: format!(
+                "Failed to open shard:{}, because of failed tables:{err_map:?}",
```

Comment:
Remove `err_map`. 
If the table fails to open, you only need to print the log and do not need to return.